lighthouse-core/report/html/templates.html

```diff
@@ -144,10 +144,85 @@
       position: relative;
       width: 100%;
     }
+
+    .lh-sticky-header {
+      --gauge-circle-size: 36px;
+      --plugin-badge-size: 18px;
+      --plugin-icon-size: 75%;
+      --score-container-width: 60px;
+      --score-number-font-size: 13px;
+      position: fixed;
+      left: 0;
+      right: 0;
+      top: var(--topbar-height);
+      font-weight: 700;
+      display: flex;
+      justify-content: center;
+      background-color: var(--color-sticky-header-bg);
+      border-bottom: 1px solid var(--color-black-200);
+      padding-top: var(--score-container-padding);
+      padding-bottom: 8px;
+      z-index: 1000;
+      pointer-events: none;
+      opacity: 0;
+    }
+
+    .lh-sticky-header.stuck {
+      pointer-events: auto;
+      opacity: 1;
+    }
+
+    .lh-sticky-header .lh-gauge__label {
+      display: none;
+    }
+
+    .lh-highlighter {
+      width: var(--score-container-width);
+      height: 1px;
+      background: var(--color-highlighter-bg);
+      position: absolute;
+      bottom: -1px;
+      left: 0;
+    }
   </style>
+  <script>
+    function handleStickyHeader() {
+      const topbarEl = document.querySelector('.lh-topbar');
+      const scoreScaleEl = document.querySelector('.lh-scorescale');
+      const stickyHeaderEl = document.querySelector('.lh-sticky-header');
+      const highlightEl = document.querySelector('.lh-highlighter');
+
+      // Show sticky header when the score scale begins to go underneath the topbar.
+      const showStickyHeader =
+        topbarEl.getBoundingClientRect().bottom - scoreScaleEl.getBoundingClientRect().top >= 0;
+      stickyHeaderEl.classList.toggle('stuck', showStickyHeader);
+
+      // Highlight mini gauge when section is in view.
+      // Use the middle of the viewport as an anchor - the closest category to the middle
+      // is the one "in view".
+      let highlightIndex = 0;
+      let highlightIndexDistance = Number.POSITIVE_INFINITY;
+      for (const [index, categoryEl] of Object.entries(document.querySelectorAll('.lh-category'))) {
```

Comment:
I think this index-based approach is what is causing the behavior I was seeing with plugin and PWA flipped. Do we want the gauges to match their appearance order in the report instead?

Reply:
just confirming you are correct in your next comment - this code will work as expected when the gauge order is as expected.